src/pages/Trade/components/TradePairOrder/index.tsx

```diff
@@ -1,38 +1,95 @@
+/* eslint-disable prefer-promise-reject-errors */
 import React, { useState, useEffect, useCallback } from 'react'
 import { Form, Input, Button, Tooltip, Divider, Popover } from 'antd'
 import { FormInstance } from 'antd/lib/form'
+import BigNumber from 'bignumber.js'
 import { useContainer } from 'unstated-next'
 import { PairList } from '../../../../utils/const'
 import TradeCoinBox from '../TradeCoinBox'
 import i18n from '../../../../utils/i18n'
 import TracePairCoin from '../TracePairCoin'
 import { PairOrderFormBox, PayMeta, OrderSelectBox, OrderSelectPopver, PairBlock } from './styled'
-import OrderContainer, { OrderStep } from '../../../../containers/order'
+import OrderContainer, { OrderStep, OrderType } from '../../../../containers/order'
 import WalletContainer from '../../../../containers/wallet'
 
 export default () => {
   const [form] = Form.useForm()
   const Wallet = useContainer(WalletContainer)
   const Order = useContainer(OrderContainer)
   const [visiblePopver, setVisiblePopver] = useState(false)
-  const { price, setPrice: priceOnChange, pay, setPay: payOnChange, receive, setStep } = Order
+  const { price, pay, setPrice, setPay, receive, setStep } = Order
   const formRef = React.createRef<FormInstance>()
-  const [disabled] = useState(false)
   const [buyer, seller] = Order.pair
 
+  // disabled button
+  const [fieldPay, setFieldPay] = useState(false)
+  const [fieldPrice, setFieldPrice] = useState(false)
+
+  const MIN_PRICE = 1 / 10 ** 10
+  const MIN_VAL = 1 / 10 ** (Order.orderType === OrderType.Buy ? 8 : 10)
+  const MIN_ORDER = Order.orderType === OrderType.Buy ? 147 : 289
+
   const changePair = () => {
     setVisiblePopver(false)
     Order.togglePair()
     form.resetFields()
   }
 
+  const checkPay = (_: any, value: string): Promise<void> => {
+    const val = new BigNumber(value)
+
+    if (Number.isNaN(parseFloat(value))) {
+      setFieldPay(false)
+      return Promise.reject(i18n.t(`trade.unEffectiveNumber`))
+    }
+
+    if (val.comparedTo(MIN_VAL) === -1) {
+      setFieldPay(false)
+      return Promise.reject(i18n.t(`trade.tooSmallNumber`))
+    }
+
+    if (val.comparedTo(Order.maxPay) === 1) {
+      setFieldPay(false)
+      return Promise.reject(i18n.t(`trade.lessThanMaxNumber`))
+    }
+
+    if (new BigNumber(Order.maxPay).minus(val).lt(MIN_ORDER)) {
+      setFieldPay(false)
+      return Promise.reject(i18n.t(`trade.insuffcientCKBBalance`))
+    }
+
+    setFieldPay(true)
+
+    return Promise.resolve()
+  }
+
+  const checkPrice = (_: any, value: string): Promise<void> => {
```

Comment:
What does the `_: any` mean?

Reply:
this is antd form component  necessary paramter required.